### PR TITLE
docs: add test instructions to the README

### PR DESCRIPTION
**Pull request by SdgJlbl**
## Summary
Adding instructions for running the test suite to the README

## Notes

I've tried my best to explain the three different modes (subprocess, docker, remote) in a succinct way, but please correct me for any mistakes or inaccuracies.

I've noted that the extra dependencies for `substra-tools` are under `test`, when it's more usual to use `dev` (and `dev` is used for both `substra` and `substrafl` so it's a small discrepancy). Nothing major, but I don't know if we want to do something about it 🤷‍♀️ 

## Please check if the PR fulfills these requirements

- [ ] If the feature has an impact on the user experience, the [changelog](https://github.com/substra/substrafl/blob/main/CHANGELOG.md) has been updated
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The commit message follows the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,6 +43,58 @@ If you need support, please either raise an issue on Github or ask on [Slack](ht
...
+Then you can run the test suites in subprocess and Docker mode: 
+
+```sh 
+make test-local
+``
+
+Please be warned that some of these tests are slow and the whole test suite might require a couple hours to complete. 
+
+If you want to try out a local deployment with Kubernetes, please follow the [installation instructions](https://docs.substra.org/en/stable/contributing/getting-started.html) provided in the doc. 
+Then you should be able to run the remote tests:
```
Comment (ThibaultFy) [inline README.md:91]:
```suggestion
The following command runs the remote tests:
```

File: README.md [hunk 2]
```diff
@@ -38,6 +43,58 @@ If you need support, please either raise an issue on Github or ask on [Slack](ht
...
+
+Now you can use the following command from `subtrafl` top level directory to run tests:
+
+```sh
+make test-subprocess
+```
+
+### Running advanced test suites
+
+Substra can be used in three different modes: using subprocesses, using Docker and using Kubernetes.
```
Comment (ThibaultFy) [inline README.md]:
```suggestion
Substra can be used in three different modes: using Python subprocesses (`subprocess`), using Docker (`docker`) and using Kubernetes (`remote`). 
```

File: README.md [hunk 3]
```diff
@@ -38,6 +43,58 @@ If you need support, please either raise an issue on Github or ask on [Slack](ht
...
+If you want to try out a local deployment with Kubernetes, please follow the [installation instructions](https://docs.substra.org/en/stable/contributing/getting-started.html) provided in the doc. 
+Then you should be able to run the remote tests:
+
+```sh
+make test-remote
+```
+
+
 
 # Appendix
```
Comment (ThibaultFy) [inline README.md:99]:
```suggestion
```
We can delete this as there is no appendix right now

File: README.md [hunk 4]
```diff
@@ -38,6 +43,58 @@ If you need support, please either raise an issue on Github or ask on [Slack](ht
...
+make test-subprocess
+```
+
+### Running advanced test suites
+
+Substra can be used in three different modes: using subprocesses, using Docker and using Kubernetes. 
+
+The command `make test-subprocess` runs the test suite in subprocess mode. It's lightweight and perfect to start. 
+
+If you want to test with the Docker mode, you will need Docker installed and running on your machine.
```
Comment (ThibaultFy) [inline README.md]:
```suggestion
To test the Docker mode, you will need Docker installed and running on your machine. 
```

**Review by ThibaultFy — CHANGES_REQUESTED** [review]
Some comments, thanks a lot for the PR, it was more than necessary !
I saw that you modify the PR, I'll finish the review after this first batch of comments :)

File: README.md (hunk 3 above)
Comment (SdgJlbl) [reply README.md:99]: There is actually the instructions for building the documentation in the appendix

File: README.md [hunk 5]
```diff
@@ -38,6 +43,58 @@ If you need support, please either raise an issue on Github or ask on [Slack](ht
...
+
+```sh
+pip install -e ".[dev]"
+```
+
+Then clone [`substra-tools`](https://github.com/Substra/substra-tools) locally, go to the top level directory of `substra-tools` and run (still in your virtual env):
+
+```sh
+pip install -e ".[test]"
+```
```
[the remark this answers is not in the document]
Comment (ThibaultFy) [reply README.md:65]: You're right, sorry for that, closing the conv

File: README.md [hunk 6]
```diff
@@ -38,6 +43,58 @@ If you need support, please either raise an issue on Github or ask on [Slack](ht
...
 Substra warmly welcomes any contribution. Feel free to fork the repo and create a pull request.
 
+## How to test
+
+You need to install `substrafl`, `substra` and `substra-tools` locally. 
+Clone this repo, then at the top level run in your virtual env:
+
+```sh
+pip install -e ".[dev]"
+``
```
[the remark this answers is not in the document]
Comment (ThibaultFy) [reply README.md:54]: up this comment

File: README.md [hunk 7]
```diff
@@ -38,6 +43,67 @@ If you need support, please either raise an issue on Github or ask on [Slack](ht
...
+
+Clone [`substra`](https://github.com/Substra/substra) locally, go to the top level directory of `substra` and run (still in your virtual env):
+
+```sh
+cd ..
+git clone git@github.com:Substra/substra.git
+cd substra
+pip install -e .
```
Comment (ThibaultFy) [inline README.md:62]: Same comment to remove `cd command`

File: README.md [hunk 8]
```diff
@@ -38,6 +43,67 @@ If you need support, please either raise an issue on Github or ask on [Slack](ht
...
+
+Install `substrafl` in editable mode with developper dependencies.
+In addition, install `substra` and `substra-tools` in editable mode.
+Clone this repo, then at the top level run in your virtual env:
+
+```sh
+pip install -e ".[dev]"
+``
+
+Clone [`substra`](https://github.com/Substra/substra) locally, go to the top level directory of `substra` and run (still in your virtual env):
```
Comment (ThibaultFy) [inline README.md:56]:
```suggestion
Clone [`substra`](https://github.com/Substra/substra) locally, go to the top level directory of `substra` and run:
```
I would add a line in the readme about virtual env, but not precise it every time

File: README.md [hunk 9]
```diff
@@ -38,6 +43,67 @@ If you need support, please either raise an issue on Github or ask on [Slack](ht
...
+
+```sh
+cd ..
+git clone git@github.com:Substra/substra.git
+cd substra
+pip install -e .
+```
+
+Then clone [`substra-tools`](https://github.com/Substra/substra-tools) locally, go to the top level directory of `substra-tools` and run (still in your virtual env):
```
Comment (ThibaultFy) [inline README.md:65]:
```suggestion
Then clone [`substra-tools`](https://github.com/Substra/substra-tools) locally, go to the top level directory of `substra-tools` and run:
```

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (SdgJlbl) [reply README.md:54]: sorry I missed it 😅

File: README.md [hunk 10]
```diff
@@ -38,6 +43,54 @@ If you need support, please either raise an issue on Github or ask on [Slack](ht
...
+
+The following command runs the test suites in subprocess and Docker mode:
+
+```sh
+make test-local
+``
+
+Please be warned that some of these tests are slow and the whole test suite might require a couple hours to complete.
+
+To try out a local deployment with Kubernetes, please follow the [installation instructions](https://docs.substra.org/en/stable/contributing/local-deployment.html) provided in the doc.
```
Comment (SdgJlbl) [inline README.md:86]: Updated the URL following https://github.com/Substra/substra-documentation/pull/257/

**Review by ThibaultFy — COMMENTED** [review]
Two very minor comments. Looks perfect this way thanks a lot !
Lets wait for https://github.com/Substra/substra-documentation/pull/257 before merging as you suggested !

File: README.md [hunk 11]
```diff
@@ -38,6 +43,54 @@ If you need support, please either raise an issue on Github or ask on [Slack](ht
...
+In addition, install `substra` and `substra-tools` in editable mode.
+It is recommended to install all the libraries in a Python virtual env.
+
+```sh
+git clone git@github.com:Substra/substrafl.git
+pip install -e "substrafl[dev]"
+git clone git@github.com:Substra/substra.git
+pip install -e substra
+git clone git@github.com:Substra/substra-tools.git
+pip install -e "substra-tools"
```
[the remark this answers is not in the document]
Comment (SdgJlbl) [reply README.md:58]: I wasn't sure how `zsh`  was handling carets 😅

**Review by ThibaultFy — APPROVED** [review]
LGTM ! :)

**Pull request merged**